eng/pipelines/docs.yml

```diff
@@ -7,75 +7,101 @@ jobs:
+        workingDirectory: $(Build.BinariesDirectory)
+        displayName: Download and Extract mdoc Zip
       - pwsh: |
           Invoke-WebRequest -MaximumRetryCount 10 -Uri "https://github.com/dotnet/docfx/releases/download/v2.43.2/docfx.zip" `
           -OutFile "docfx.zip" | Wait-Process; Expand-Archive -Path "docfx.zip" -DestinationPath "./docfx/"
         workingDirectory: $(Build.BinariesDirectory)
         displayName: Download and Extract DocFX
+      - pwsh: |
+          dotnet restore "$(Build.SourcesDirectory)\eng\docgeneration\assets\docgen.csproj"
+          dotnet restore "$(Build.SourcesDirectory)\eng\service.proj"
+        displayName: Restore Nuget Packages Needed
+      - pwsh: |
+          New-Item -Type dir dll-docs/my-api
+          New-Item -Type dir dll-docs/dependencies/my-api
+          New-Item -Type dir dll-xml-output
+          New-Item -Type dir dll-yaml-output
+          New-Item -Type dir docfx-output
+        displayName: Create Directories for Doc Generation
+        workingDirectory: $(Build.BinariesDirectory)
+      - task: DownloadPipelineArtifact@2
+        displayName: 'Download published packages'
+        inputs:
+          buildType: specific
+          project: '590cfd2a-581c-4dcb-a12e-6568ce786175'
+          definition: 37
+          buildVersionToDownload: latestFromBranch
+          tags: Scheduled
+          artifactName: packages
+          targetPath: '$(System.ArtifactsDirectory)/packages'
+      - pwsh: ". '$(Build.SourcesDirectory)/eng/docgeneration/script/docgen.ps1';ExtractDlls -PkgDirPath $(System.ArtifactsDirectory)/packages -DllDirPath $(Build.BinariesDirectory)/dll-docs/my-api"
+        displayName: Extract dlls from packages and copy then to dlls directory
+      - task: CopyFiles@2
+        displayName: 'Copy over Api Dependencies'
+        inputs:
+          SourceFolder: '$(Build.SourcesDirectory)/restoredPackages/netstandard.library/2.0.0/build/netstandard2.0/ref'
+          TargetFolder: '$(Build.BinariesDirectory)/dll-docs/dependencies/my-api'
       - task: CmdLine@2
-        displayName: Provision DocFX Directory
+        displayName: Initialize Frameworks File
         inputs:
-          script: $(Build.BinariesDirectory)/docfx/docfx.exe init -q
-          workingDirectory: $(Build.SourcesDirectory)
+          script: mdoc.exe fx-bootstrap ../dll-docs
+          workingDirectory: $(Build.BinariesDirectory)/mdoc
           failOnStderr: true
-      - pwsh: ls
-        workingDirectory: $(Build.SourcesDirectory)/docfx_project/
       - pwsh: |
-          Copy-Item "$(Build.SourcesDirectory)/eng/docgeneration/docfx.json" -Destination "$(Build.SourcesDirectory)/docfx_project/" -Force
-        displayName: Copy over docfx.json
+          Copy-Item -Path "$(Build.SourcesDirectory)/restoredPackages/microsoft.bcl.asyncinterfaces/1.0.0-preview6.19303.8/ref/netstandard2.1/Microsoft.Bcl.AsyncInterfaces.dll" `
```

Comment:
We definitely should not make the assumptions that these packages are restored into the restoredPackages folder that is  a bad assumption. Instead we should write a custom target in one of the projects you are restoring that will copy the files.  See what I did https://dev.azure.com/azure-sdk/internal/_git/azure-sdk-build-tools?path=%2Ftools%2FRestoreToolPackages.proj&version=GBmaster as an example.